api/client/proxy/client.go

```diff
@@ -198,23 +212,101 @@ func NewClient(ctx context.Context, cfg ClientConfig) (*Client, error) {
 	}
 
 	clt, sshErr := newSSHClient(ctx, &cfg)
-	if sshErr == nil {
-		return clt, nil
+	// Only aggregate errors if there was an issue dialing the grpc server so
+	// that helpers like trace.IsAccessDenied will still work.
+	if grpcErr == nil {
+		return clt, trace.Wrap(sshErr)
```

Comment:
Good point, and the problem is really just that trace aggregate does not implement `Unwrap`.

`errors.Is` works for checking if an error is of a specific value, which includes type but does a more specific comparison on the error value.
`errors.As` works for finding a type of error in the aggregate error chain

I think our naming convention `trace.IsFoo`, is a little confusing since it's a type check like `errors.As`, but that's probably just because trace predates a lot of the standard lib error changes.

Leveraging errors.As works, for example:
```go
// IsAccessDenied detects if this error is of AccessDeniedError type
func IsAccessDenied(err error) bool {
	var asErr *AccessDeniedError
	return errors.As(err, &asErr)
}
```

Reply:
aggregate.As walks all the errors already:
```go
// As implements the `As` interface, by iterating through each error in the
// aggregate and invoking `errors.As`.
func (r aggregate) As(t interface{}) bool {
	for _, err := range r {
		if errors.As(err, t) {
			return true
		}
	}
	return false
}
```